ddtrace/tracer.py

```diff
@@ -504,15 +519,26 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
 
         return span
 
-    def _start_runtime_worker(self):
+    def _init_runtime_worker(self):
         if not self._dogstatsd_url:
             return
 
         self._runtime_worker = RuntimeWorker(self._dogstatsd_url)
+        self._init_runtime_worker_called = True
         # force an immediate update constant tags since we have reset services
         # and generated a new runtime id
         self._runtime_worker.update_runtime_tags()
 
+    def _init_agent_writer(self):
+        self.writer = AgentWriter(
+            self._agent_url,
+            sampler=self.sampler,
+            priority_sampler=self.priority_sampler,
+            dogstatsd=get_dogstatsd_client(self._dogstatsd_url),
+            report_metrics=config.health_metrics_enabled,
+        )
+        self._init_agent_writer_called = True
```

Comment:
can we just use `self.writer` as a guess?
Adding yet another state variable is just adding more mess at this stage 😢

Reply:
Unfortunately because of how `configure()` currently does not instantiate a new tracer and hence we can't be certain that writer has not already been set previously, e.g., multiple calls to `configure()`. We could have a complicated check that ensures that the necessary changes to the writer weren't made if one exists but having a state that we can check to see if the init function was called makes this easier to manage, at least until we make the upcoming changes to the `configure()`.